holoviews/plotting/util.py

```diff
@@ -295,11 +296,43 @@ def attach_streams(plot, obj):
     Attaches plot refresh to all streams on the object.
     """
     def append_refresh(dmap):
-        for stream in dmap.streams:
+        for stream in get_streams(dmap):
             stream._hidden_subscribers.append(plot.refresh)
     return obj.traverse(append_refresh, [DynamicMap])
 
 
+def get_sources(obj, index=None):
+    """
+    Traverses Callable graph to resolve sources on
+    DynamicMap objects, returning a list of sources
+    indexed by the Overlay layer.
+    """
```

Comment:
Can't this be simplified to:

```python
if isinstance(obj, DynamicMap) and isinstance(obj.callback, Callable):
    return [(index, obj)]
```

leaving the rest of the function to handle ``DynamicMaps`` with ``Callable`` callbacks?